components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -482,18 +482,22 @@ abstract class AbstractFetchDownloadService : Service() {
         uri: Uri?,
         referer: Uri?
     ) {
-        context.addCompletedDownload(
-            title = title,
-            description = description,
-            isMediaScannerScannable = isMediaScannerScannable,
-            mimeType = mimeType,
-            path = path,
-            length = length,
-            // Only show notifications if our channel is blocked
-            showNotification = showNotification,
-            uri = uri,
-            referer = referer
-        )
+        try {
+            context.addCompletedDownload(
+                title = title,
+                description = description,
+                isMediaScannerScannable = isMediaScannerScannable,
+                mimeType = mimeType,
+                path = path,
+                length = length,
+                // Only show notifications if our channel is blocked
+                showNotification = showNotification,
+                uri = uri,
+                referer = referer
+            )
+        } catch (e: IllegalArgumentException) {
```

Comment:
Reading the code a bit, this is coming out of a ContentResolver - as its trying to resolve a ContentProvider for the `downloads` authority (and, fails, obviously, it doesn't seem like a well-formed authority to me). But, it only does that (attempts to resolve a content provider via authority) because the scheme is `content`.

And just above, we already have a check to ensure that url's scheme is `http` or `https` - `val url = if (!download.isScheme(listOf("http", "https"))) null else download.url.toUri()`.

So, it seems like this shouldn't be possible to hit this..? Perhaps the crash reports are before that url scheme check was added? But no - this check was added in https://github.com/mozilla-mobile/android-components/pull/7280 which was 2 years ago, and we've reports from recent versions (sentry says, A-C 98 hit this somehow... e.g. https://sentry.prod.mozaws.net/operations/firefox-nightly/issues/10551720/events/latest/).

So, how can this be..?

nit: i'd also move that `url` check into this method, where we're actually calling into `DownloadManager`.

Reply:
@Amejia481 @grigoryk The way I read this is that these are different URLs! Our check here is for the original HTTP URL of the download, but the failing check / crash is for the base URL of all downloads i.e., the URL of the table to insert the download into: https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/content/ContentResolver.java;l=2189?q=ContentResolver.java

Reply:
The original download URL is passed along as content values, but the exception is thrown for because `Downloads.Impl.CONTENT_URI` doesn't resolve on those devices: https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/app/DownloadManager.java;l=1569

Reply:
Thanks for the clarification, in that case how should we recover from that or prevent that behavior? At the moment catching the exception prevent the crash, I'm not sure how we could proceed?